plugins/modules/ec2_transit_gateway.py

```diff
@@ -295,35 +279,35 @@ def wait_for_status(self, wait_timeout, tgw_id, status, skip_deleted=True):
                     if transit_gateway.get("state") == status:
                         status_achieved = True
                         break
-
                     elif transit_gateway.get("state") == "failed":
                         break
-
                 else:
                     sleep(polling_increment_secs)
 
-            except ClientError as e:
+            except AnsibleEC2Error as e:
                 self._module.fail_json_aws(e)
 
         if not status_achieved:
             self._module.fail_json(msg="Wait time out reached, while waiting for results")
 
         return transit_gateway
 
-    def get_matching_tgw(self, tgw_id, description=None, skip_deleted=True):
-        """search for  an existing tgw by either tgw_id or description
-        :param tgw_id:  The AWS id of the transit gateway
-        :param description:  The description of the transit gateway.
-        :param skip_deleted: ignore deleted transit gateways
-        :return dict: transit gateway object
+    def get_matching_tgw(
+        self, tgw_id: Optional[str], description: Optional[str] = None, skip_deleted: bool = True
+    ) -> Optional[Dict[str, Any]]:
+        """Search for an existing tgw by either tgw_id or description.
+        :param tgw_id: The AWS id of the transit gateway.
+        :param description: The description of the transit gateway.
+        :param skip_deleted: Ignore deleted transit gateways.
+        :return: Transit gateway object.
         """
         filters = []
         if tgw_id:
             filters = ansible_dict_to_boto3_filter_list({"transit-gateway-id": tgw_id})
 
         try:
-            response = AWSRetry.exponential_backoff()(self._connection.describe_transit_gateways)(Filters=filters)
-        except (ClientError, BotoCoreError) as e:
+            response = describe_ec2_transit_gateways(self._connection, Filters=filters)
+        except AnsibleEC2Error as e:
```

Comment:
will this not be taken care of by a.aws' module_utils.ec2?

Reply:
except block has been removed as its not required.
describe_ec2_transit_gateways being called here is the one in a.aws's module_utils/ec2